plugins/pay.c

```diff
@@ -372,24 +419,56 @@ static struct command_result *getroute_done(struct command *cmd,
 	feepercent = ((double)fee) * 100.0 / ((double) pc->msatoshi);
 
 	if (fee > pc->exemptfee && feepercent > pc->maxfeepercent) {
+		const jsmntok_t *charger;
+
 		attempt_failed_fmt(pc, "{ 'message': 'Route wanted fee of %"PRIu64" msatoshis' }", fee);
 
+		/* Remember this if we eliminating this causes us to have no
+		 * routes at all! */
+		if (!pc->expensive_route)
+			pc->expensive_route
+				= tal_fmt(pc, "Route wanted fee of %"PRIu64
+					  " msatoshis", fee);
+
+		/* Try excluding most fee-charging channel (unless it's in
+		 * routeboost). */
+		charger = find_worst_channel(buf, t, "msatoshi", pc->msatoshi);
```

Comment:
This is applied to the route that failed our constraints, right? The `msatoshi` values in the route however are absolute, not deltas, so if I'm not mistaken this will always return the first channel since that's the one that has the maximum cumulative value, or am I missing something here?

Reply:
I think the internal logic of find_worst_channel gets this right:

		if (worst == NULL || val - prev > worstval) {
			worst = route;
			worstval = val - prev;
		}

Reply:
Oh, I missed the comparison with `prev` that's ok then.